server/src/handlers/crawl_handler.rs

```diff
@@ -0,0 +1,184 @@
+#[utoipa::path(
+    put,
+    path = "/crawl",
+    context_path = "/api",
+    tag = "Dataset",
+    request_body(content = UpdateCrawlReqPayload, description = "JSON request payload to update a crawl", content_type = "application/json"),
+    responses(
+        (status = 200, description = "Crawl updated successfully", body = CrawlRequest),
+        (status = 400, description = "Service error relating to updating the dataset", body = ErrorResponseBody),
+    ),
+    params(
+        ("TR-Dataset" = uuid::Uuid, Header, description = "The dataset id to use for the request"),
+    ),
+    security(
+        ("ApiKey" = ["admin"]),
+    )
+)]
+pub async fn update_crawl_request(
+    data: web::Json<UpdateCrawlReqPayload>,
+    pool: web::Data<Pool>,
+    broccoli_queue: web::Data<BroccoliQueue>,
+    dataset_org_plan_sub: DatasetAndOrgWithSubAndPlan,
+    _user: AdminOnly,
+) -> Result<HttpResponse, ServiceError> {
+    let data = data.into_inner();
+
+    let crawl = update_crawl_query(
+        data.crawl_options.clone(),
+        data.crawl_id,
+        dataset_org_plan_sub.dataset.id,
+        pool.clone(),
+        broccoli_queue.clone(),
+    )
+    .await?;
+
+    Ok(HttpResponse::Ok().json(crawl))
+}
+
+/// Delete a crawl request
+///
+/// This endpoint is used to delete an existing crawl request for a dataset. The request payload should contain the crawl id to delete.
+#[utoipa::path(
+    delete,
+    path = "/crawl/:crawl_id",
+    context_path = "/api",
+    tag = "Dataset",
+    responses(
+        (status = 204, description = "Crawl deleted successfully"),
+        (status = 400, description = "Service error relating to deleting the dataset", body = ErrorResponseBody),
+    ),
+    params(
+        ("TR-Dataset" = uuid::Uuid, Header, description = "The dataset id to use for the request"),
+        ("crawl_id" = uuid::Uuid, Path, description = "The id of the crawl to delete"),
+    ),
+    security(
+        ("ApiKey" = ["admin"]),
+    )
+)]
+pub async fn delete_crawl_request(
+    data: web::Path<uuid::Uuid>,
+    pool: web::Data<Pool>,
+    _user: AdminOnly,
+) -> Result<HttpResponse, ServiceError> {
+    let crawl_id = data.into_inner();
+
+    delete_crawl_query(crawl_id, pool.clone()).await?;
+
+    Ok(HttpResponse::NoContent().finish())
+}
+
+#[derive(Debug, Serialize, Deserialize, ToSchema, Clone)]
+pub struct GetCrawlRequestsReqPayload {
+    pub page: Option<i64>,
+    pub limit: Option<i64>,
+}
+
+/// Get all crawl requests for a dataset
+///
+/// This endpoint is used to get all crawl requests for a dataset.
+#[utoipa::path(
+    post,
+    path = "/crawl/dataset",
+    context_path = "/api",
+    tag = "Dataset",
+    request_body(content = GetCrawlRequestsReqPayload, description = "JSON request payload to get all crawl requests", content_type = "application/json"),
```

Comment:
We should not have request bodies for get requests. These should be URL query params.